demos/node-classification/deep-graph-info-max/gcn-deep-graph-info-max-cora-example.ipynb

```diff
@@ -0,0 +1,284 @@
+      "text/plain": [
+       "<Figure size 504x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "es = EarlyStopping(monitor=\"loss\", min_delta=0, patience=20)\n",
+    "history = model.fit(gen, epochs=1000, verbose=0, callbacks=[es])\n",
```

Comment:
FWIW, this fails with tensorflow 2.0.0 with `TypeError: int() argument must be a string, a bytes-like object or a number, not 'tuple'`. (I didn't try 2.0.1.)

I think it might be somehow related to the shape of the tuples/tensors returned by `__getitem__` below, because I saw something similar with `KGTripleGenerator`.

This is a broader problem (#614), and may affect other models too, so is potentially not something we need to fix here... but we should at least have an issue about "some models fail with tensorflow 2.0.0" to deal with in some form before the next release.

<details><summary>Full stack trace</summary>

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-2539c1ce905f> in <module>
      1 es = EarlyStopping(monitor="loss", min_delta=0, patience=20)
----> 2 history = model.fit(gen, epochs=1000, verbose=0, callbacks=[es])
      3 plot_history(history)

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/training.py in fit(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_freq, max_queue_size, workers, use_multiprocessing, **kwargs)
    726         max_queue_size=max_queue_size,
    727         workers=workers,
--> 728         use_multiprocessing=use_multiprocessing)
    729 
    730   def evaluate(self,

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/training_v2.py in fit(self, model, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_freq, **kwargs)
    222           validation_data=validation_data,
    223           validation_steps=validation_steps,
--> 224           distribution_strategy=strategy)
    225 
    226       total_samples = _get_total_number_of_samples(training_data_adapter)

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/training_v2.py in _process_training_inputs(model, x, y, batch_size, epochs, sample_weights, class_weights, steps_per_epoch, validation_split, validation_data, validation_steps, shuffle, distribution_strategy, max_queue_size, workers, use_multiprocessing)
    545         max_queue_size=max_queue_size,
    546         workers=workers,
--> 547         use_multiprocessing=use_multiprocessing)
    548     val_adapter = None
    549     if validation_data:

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/training_v2.py in _process_inputs(model, x, y, batch_size, epochs, sample_weights, class_weights, shuffle, steps, distribution_strategy, max_queue_size, workers, use_multiprocessing)
    604       max_queue_size=max_queue_size,
    605       workers=workers,
--> 606       use_multiprocessing=use_multiprocessing)
    607   # As a fallback for the data type that does not work with
    608   # _standardize_user_data, use the _prepare_model_with_inputs.

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/data_adapter.py in __init__(self, x, y, sample_weights, shuffle, **kwargs)
    611         yield x[i]
    612     dataset = dataset_ops.DatasetV2.from_generator(generator, nested_dtypes,
--> 613                                                    output_shapes=nested_shape)
    614     if shuffle:
    615       dataset = dataset.shuffle(len(x))

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/data/ops/dataset_ops.py in from_generator(generator, output_types, output_shapes, args)
    538     else:
    539       output_shapes = nest.map_structure_up_to(
--> 540           output_types, tensor_shape.as_shape, output_shapes)
    541     if args is None:
    542       args = ()

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/data/util/nest.py in map_structure_up_to(shallow_tree, func, *inputs)
    469                          for input_tree in inputs]
    470 
--> 471   results = [func(*tensors) for tensors in zip(*all_flattened_up_to)]
    472   return pack_sequence_as(structure=shallow_tree, flat_sequence=results)

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/data/util/nest.py in <listcomp>(.0)
    469                          for input_tree in inputs]
    470 
--> 471   results = [func(*tensors) for tensors in zip(*all_flattened_up_to)]
    472   return pack_sequence_as(structure=shallow_tree, flat_sequence=results)

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/framework/tensor_shape.py in as_shape(shape)
   1214     return shape
   1215   else:
-> 1216     return TensorShape(shape)
   1217 
   1218 

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/framework/tensor_shape.py in __init__(self, dims)
    774       else:
    775         # Got a list of dimensions
--> 776         self._dims = [as_dimension(d) for d in dims_iter]
    777 
    778   @property

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/framework/tensor_shape.py in <listcomp>(.0)
    774       else:
    775         # Got a list of dimensions
--> 776         self._dims = [as_dimension(d) for d in dims_iter]
    777 
    778   @property

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/framework/tensor_shape.py in as_dimension(value)
    716     return value
    717   else:
--> 718     return Dimension(value)
    719 
    720 

~/.pyenv/versions/sg/lib/python3.6/site-packages/tensorflow_core/python/framework/tensor_shape.py in __init__(self, value)
    191       raise TypeError("Cannot convert %s to Dimension" % value)
    192     else:
--> 193       self._value = int(value)
    194       if (not isinstance(value, compat.bytes_or_text_types) and
    195           self._value != value):

TypeError: int() argument must be a string, a bytes-like object or a number, not 'tuple'
```

</details>

Reply:
> but we should at least have an issue about "some models fail with tensorflow 2.0.0" to deal with in some form before the next release

https://github.com/stellargraph/stellargraph/issues/1045